packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ScmTypeFactory.cs

```diff
@@ -6,14 +6,59 @@
 using System.IO;
 using System.Linq;
 using System.Net;
+using Microsoft.Generator.CSharp.ClientModel.Providers;
 using Microsoft.Generator.CSharp.Input;
 using Microsoft.Generator.CSharp.Providers;
 
 namespace Microsoft.Generator.CSharp.ClientModel
 {
     public class ScmTypeFactory : TypeFactory
     {
-        private Dictionary<InputOperation, MethodProviderCollection?> _operations = new Dictionary<InputOperation, MethodProviderCollection?>();
+        private readonly IDictionary<InputModelType, ModelProvider> _models = new Dictionary<InputModelType, ModelProvider>();
+        private readonly IDictionary<InputEnumType, EnumProvider> _enums = new Dictionary<InputEnumType, EnumProvider>();
+        private readonly IDictionary<InputClient, ClientProvider> _clients = new Dictionary<InputClient, ClientProvider>();
+        private readonly Dictionary<InputOperation, MethodProviderCollection?> _operations = new Dictionary<InputOperation, MethodProviderCollection?>();
+
+        public override ModelProvider CreateModel(InputModelType inputModel)
```

Comment:
Do you think it is OK that we have this return type as `TypeProvider` in MGC but override it to `ModelProvider` in SCM?
I did not do this intentionally but it turns out to look fine to me

Reply:
I don't think we should force people to override our ModelProvider if they want to extend MGC.ClientModel

Reply:
This is fixed